include/gadget/filter.h

```diff
@@ -0,0 +1,128 @@
+/* SPDX-License-Identifier: (GPL-2.0 WITH Linux-syscall-note) OR Apache-2.0 */
+
+#ifndef FILTER_H
+#define FILTER_H
+
+#include <bpf/bpf_helpers.h>
+#include <gadget/macros.h>
+#include <gadget/types.h>
+#include <gadget/mntns_filter.h>
+
+#define GADGET_INVALID_ID ((uid_t)-1)
+
+// user space terminology
+const volatile gadget_pid targ_pid = 0;
+GADGET_PARAM(targ_pid);
+
+const volatile gadget_tid targ_tid = 0;
+GADGET_PARAM(targ_tid);
+
+const volatile gadget_uid targ_uid = GADGET_INVALID_ID;
+GADGET_PARAM(targ_uid);
+
+const volatile gadget_gid targ_gid = GADGET_INVALID_ID;
+GADGET_PARAM(targ_gid);
+
+const volatile gadget_comm targ_comm[TASK_COMM_LEN] = {};
+GADGET_PARAM(targ_comm);
+
+static __always_inline bool gadget_should_discard_pid_tid(gadget_pid pid,
+							  gadget_tid tid)
+{
+	if (targ_pid != 0 && targ_pid != pid)
+		return true;
+
+	if (targ_tid != 0 && targ_tid != tid)
+		return true;
+
+	return false;
+}
+
+static __always_inline bool gadget_should_discard_uid_gid(gadget_uid uid,
+							  gadget_gid gid)
+{
+	if (targ_uid != GADGET_INVALID_ID && targ_uid != uid)
+		return true;
+
+	if (targ_gid != GADGET_INVALID_ID && targ_gid != gid)
+		return true;
+
+	return false;
+}
+
+static __always_inline bool
+gadget_should_discard_comm(gadget_comm comm[TASK_COMM_LEN])
+{
+	if (targ_comm[0] == '\0')
+		return false;
+
+	for (int i = 0; i < TASK_COMM_LEN; i++) {
+		if (comm[i] != targ_comm[i])
+			return true;
+	}
+
+	return false;
+}
+
+// gadget_should_discard returns true if the gadget should skip this event.
+static __always_inline bool
+gadget_should_discard(gadget_mntns_id mntns_id, gadget_pid pid, gadget_tid tid,
+		      gadget_comm comm[TASK_COMM_LEN], gadget_uid uid,
+		      gadget_uid gid)
+{
+	if (gadget_should_discard_mntns_id(mntns_id))
+		return true;
+
+	if (gadget_should_discard_pid_tid(pid, tid))
+		return true;
+
+	if (gadget_should_discard_uid_gid(uid, gid))
+		return true;
+
+	if (gadget_should_discard_comm(comm))
+		return true;
+
+	return false;
```

Comment:
> I am wondering if we cannot think of having an array of filters (array of function pointers).
This function would loop the function pointers, so it would ease maintaining.
Maybe a macro would be better, but maybe there is no silver bullet here.

I'm not sure how it'll help and how it plays out with the verifier, so I'd prefer to keep it simple.